Symfony 6 support

loic425 opened the pull request:
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | master          |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | hope no                                                       |
| Deprecations?   | no |
| Related tickets | fixes #13274                     |
| License         | MIT                                                          |

<!--
 - Bug fixes must be submitted against the 1.10 or 1.11 branch(the lowest possible)
 - Features and deprecations must be submitted against the master branch
 - Make sure that the correct base branch is set

 To be sure you are not breaking any Backward Compatibilities, check the documentation:
 https://docs.sylius.com/en/latest/book/organization/backward-compatibility-promise.html
-->

loic425 committed:
Allow Symfony 6 and prepare CI

loic425 committed:
Fix specifications for Addressing bundle

loic425 committed:
Fix attribute bundle test app

loic425 committed:
Fix specs for province address constraint validator

loic425 committed:
Fix Channel bundle test app

loic425 committed:
Trying to remove Mocker container from test app

loic425 committed:
Fix Currency bundle test app

loic425 committed:
Fix Inventory bundle test app

loic425 committed:
Fix Locale bundle test app

loic425 committed:
Fix Money bundle test app

loic425 committed:
Fix Payment bundle test app

loic425 committed:
Fix Customer bundle test app

loic425 committed:
Fix specifications for build product variant form subscriber

loic425 committed:
Fix Product bundle test app

loic425 committed:
Fix generate product variants subscriber

loic425 committed:
Fix DI

loic425 committed:
Fix specifications for register calculators pass

loic425 committed:
Fix Taxonomy bundle test app

loic425 committed:
Fix specifications for security controller

loic425 committed:
Fix PHPUnit tests for Ui bundle

loic425 committed:
Apply suggestions from code review

Co-authored-by: Łukasz Chruściel <lchrusciel@gmail.com>

loic425 committed:
Fix taxation test app

loic425 committed:
Apply suggestions from code review

Co-authored-by: Łukasz Chruściel <lchrusciel@gmail.com>

loic425 committed:
Fix product variant generator

loic425 committed:
Fix addressing test app

loic425 committed:
Fix PHPUnit tests

loic425 committed:
Trying to enable authenticator manager

loic425 committed:
Trying to fix Symfony 6 dependencies for Shipping bundle

loic425 committed:
Fix test app

loic425 committed:
Fix services configuration for deprecations

loic425 committed:
Fix package name on deprecations

loic425 committed:
Fix Promotion bundle for Symfony 6

loic425 committed:
Fix Order bundle

loic425 committed:
Fix user component

loic425 committed:
Use encoder aware interface from our polyfill

loic425 committed:
Add missing method get user identifier

loic425 committed:
Fix user bundle

loic425 committed:
Allow oauth bundle 2.0 beta

loic425 committed:
Fix tests app

loic425 committed:
Use UserNotFound exception

loic425 committed:
Use UserNotFound exception

loic425 committed:
Fix some Phpspec tests

loic425 committed:
Fix Core component

loic425 committed:
Fix Core bundle

loic425 committed:
Fix dependencies

loic425 committed:
Fix Phpspec tests

loic425 committed:
Fix CI configuration

loic425 committed:
Fix user component dependencies

loic425 committed:
Fix PHPUnit tests

loic425 committed:
Fix Review bundle

loic425 committed:
Remove SwiftMailer from test app kernel

loic425 committed:
Fix config for test app

loic425 committed:
Fix security config on test app

loic425 committed:
Fix test app kernel

HeahDude commented:
There will be BC breaks if Symfony 6 is allowed in a new minor without explicit upgrade request from projects based on it (master seems to be targeted without bump https://github.com/Sylius/Sylius/pull/13969/files#diff-035b723a6c827cf63f13f32f9ce11dbaf725c2124dd6a43d0138cb4b42f75054R50).

IMHO a new major is needed to let people opt-in into this upgrade (e.g when Symfony 5.4 deprecations are dealt with, and some Sylius deprecations as well?).
We should not just let composer choose between 5.4 and 6.0 by upgrading a minor version of Sylius.

HeahDude commented:
Since Symfony Flex is a requirement, a solution could be to leverage the extra `require` config option.
Defining it to `5.4.*` in `composer.json` in this PR would allow to let people opt-in and prevent breaking in a minor update.

loic425 committed:
Apply suggestions from code review

lchrusciel commented:
Hey Jules,

thanks for raising concerns. Sylius itself does not require Symfony flex, but we've added it to Sylius-Standard a long time ago. We would probably go with the minor release for Sf6, as we did previously, but people are able to stay with Sf5.4 as you mentioned. However, we will consider a change of this approach for future releases

loic425 committed:
Use our polyfill for UserNotFoundException

loic425 committed:
Trying to use data mapper

lchrusciel committed:
[Fix][Maintenance] Cart item data mapper adjusted to be Sf6 compatible

loic425 committed:
Trying to remove Swiftmailer from test app

loic425 committed:
Allow sylius/theme-bundle symfony-6 branch

loic425 committed:
Fix Phpspec tests

loic425 committed:
Fix some deprecations

loic425 committed:
Fix some PHPUnit tests

loic425 committed:
[PayumBundle] Add support for Symfony 6

loic425 committed:
Checking package versions

loic425 committed:
Fix packages versions

lchrusciel committed:
[Core] Add test for symfony mailer adapter and improve testing app

lchrusciel committed:
[GH] Configure testing for CoreBundle with Swiftmailer

loic425 committed:
Fix packages versions for doctrine/dbal

loic425 committed:
Upgrade main packages to allow Symfony 6

loic425 committed:
Do not remove conflict but adjust versions

lchrusciel committed:
[Sf6] Add UserBundle mappings to Symfony6 support

Chris53897 commented:
@loic425 Is it possible that you update `sylius/theme-bundle`: "^2.1.1", to "2.3.x-dev".
As this is a blocker for some symfony 6 components. https://packagist.org/packages/sylius/theme-bundle#v2.2.0
I do not know if there are more.

Zales0123 committed:
[ApiBundle] Fix test app configuration and mailer-based tests

Zales0123 committed:
[ApiBundle] Fix specs

Zales0123 committed:
[ApiBundle] Fix Promotion-related phpunit tests

Zales0123 committed:
[ApiBundle] Test one case with swiftmailer enabled

NoResponseMate committed:
[AdminBundle] Ensure it works with swiftmailer and symfony mailer

NoResponseMate committed:
[AdminBundle][Tests] Use the new directory structure

NoResponseMate committed:
[AdminBundle] Remove flash bag and session services usage

TheMilek committed:
[ShopBundle] Fix spec tests

NoResponseMate committed:
[AdminBundle] Bring back and deprecate session and flash bag usage

Zales0123 committed:
Bump knplabs/gaufrette dependency

Zales0123 committed:
Delete unneeded swiftmailer file

Zales0123 committed:
Hot-fix with symfony mockery container

Zales0123 committed:
Bump mailer bundle dependencies

Zales0123 committed:
Make all dependencies consistent

Zales0123 committed:
Update some Sylius packages dependencies

Zales0123 committed:
Remove unneedd conflicts

Zales0123 committed:
Fix static checks for Symfony 5.4

Zales0123 committed:
[WIP] Fix static checks on Symfony 6

ernestWarwas committed:
[Symfony 6] Fix application configuration

NoResponseMate committed:
Fix static checks on symfony 6 - continuation

ernestWarwas committed:
[Doctrine] cache provider fix

[Doctrine] cache provider fix

code style

NoResponseMate committed:
[Test] Refactor EmailChecker to work with Symfony Mailer

NoResponseMate committed:
Use Symfony Mailer by default

NoResponseMate committed:
[Maintenance] Update email assertion in Unit tests

NoResponseMate committed:
[Maintenance][Admin] Add FlysystemBundle to the test app

NoResponseMate committed:
[Maintenance] Transform email html content to text
Fix possible false-negative case in EmailChecker

NoResponseMate committed:
[Maintenance] Remove Swiftmailer-based unit tests

NoResponseMate committed:
[Maintenance] Updated mailer config

NoResponseMate committed:
[Maintenance] Move EmailChecker from Core into Behat

Rafikooo committed:
[Maintenance] Support both UserPasswordEncoder & UserPasswordHasher

ernestWarwas committed:
[BC] Bc layer for request stack and session interface added

NoResponseMate committed:
[Maintenance] Fix RedirectPathExtensionTest

NoResponseMate committed:
[Maintenance] Update routings' `controller` node syntax

Rafikooo committed:
[Maintenance] PasswordUpdater supports both Symfony 5.4 & Symfony 6

NoResponseMate committed:
[Behat] Fix SecurityService to work with new session management flow

Rafikooo committed:
[Maintenance] psalm.xml updated with deprecated class and interface

Zales0123 committed:
[WIP] Fix PHPUnit tests with Symfony 6 and new session management

Zales0123 committed:
Finish fixes in PHPUnit tests

Zales0123 committed:
Fix container bugs

Zales0123 committed:
Skip tests that cannot be fixed now

Zales0123 committed:
Extract session management in tests

Rafikooo committed:
[GithubActions] Show diff in comment to PR job removed

NoResponseMate committed:
[Behat] Add session hook context

NoResponseMate committed:
[Behat] Add session hook context

Zales0123 committed:
Revert SyliusLabs/PolyfilSymfonySecurity fork branch

NoResponseMate committed:
[Maintenance][Behat] Temporarily allow failing cli tests

NoResponseMate committed:
[Maintenance] Bring back swiftmailer-based tests in packages

NoResponseMate committed:
[Maintenance] Fix test apps configuration for swiftmailer env

Zales0123 committed:
Temporarily run only features/account/ Behat scenarios in Symfony 6

Zales0123 committed:
Fix Behat scenarios for User account

Zales0123 committed:
Composer.json fixes

Zales0123 committed:
Remove some deprecations to fix most of Behat scenarios

Zales0123 committed:
Do not reset token storage between tests

Zales0123 committed:
Make IriConverter stateless

Zales0123 committed:
Return the whole application build

Zales0123 committed:
Return temporarily removed workflows

Zales0123 committed:
Remove duplicated code in CachedRouteNameResolver

Zales0123 committed:
Remove duplicated code in Symfony6PrivateServicesPass

Rafikooo committed:
[ProductAttribute] The way of processing query parameters changed

Rafikooo committed:
[GithubActions] Redundand Run JS Behat command removed

Zales0123 committed:
[Symfony 6] Remove setting services for test as public

Rafikooo committed:
[Forms] Legacy error messages disabled

Rafikooo committed:
[Promotions][Behat] Form validation message changed according to the new messaging system

Rafikooo committed:
[Behat] Unused step implementation removed

Zales0123 committed:
Fix packages versions

Zales0123 committed:
Remove unneeded ApiPlatform version check in static checks

Rafikooo committed:
[CatalogPromotions][API][Behat] Step with a different behavior in SF-5.4 and SF-6 changed

Rafikooo committed:
[Behat][API][Admin] Proper IRIs in the admin context

Zales0123 committed:
Fix encore build for AdminBundle and CoreBundle

Zales0123 committed:
Fix deprecated controller syntax to fix the build

Zales0123 committed:
[Github Actions] Make timeout for non-JS build longer

GSadee committed:
bug #14414 [Github Actions] Make timeout for non-JS build longer (Zales0123)

This PR was merged into the symfony-6 branch.

Discussion
----------

| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | symfony-6  |
| Bug fix?        | yes                                                       |
| New feature?    | no |
| BC breaks?      | no |
| Deprecations?   | no |
| Related tickets |  |
| License         | MIT                                                          |

On the recent builds, we've experienced a lot of cancelled jobs due to the longer build time. I raise the timeout to 35 minutes, although we should think about fixing the problem rather than overpassing it 🖖 

Commits
-------

d321bee084667631b9263ef57542aeb127569102 [Github Actions] Make timeout for non-JS build longer

coldic3 committed:
bug #14407 [Symfony 6] Remove setting services for test as public (Zales0123)

This PR was merged into the symfony-6 branch.

Discussion
----------

| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | symfony-6                  |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets | mentioned in [this comment](https://github.com/Sylius/Sylius/pull/14376#discussion_r985750541)  |
| License         | MIT                                                          |


Commits
-------

a49d879e3d62b89ebcc74f3f7787e0983cb8265f [Symfony 6] Remove setting services for test as public

coldic3 reviewed (approved):
I did a small QA on Sylius-Standard and didn't find any issues 🎉

Zales0123 commented on src/Sylius/Bundle/CoreBundle/Resources/config/services.xml:
```diff
@@ -269,8 +269,6 @@
             <argument type="service" id="sylius.email_sender" />
         </service>
 
-        <service id="security.authentication_manager" alias="security.authentication.manager" />
```
I believe we should mention it in the UPGRADE file 🖖

Zales0123 reviewed (approved):
I've also checked it out on the Sylius-Standard and it seems to work 🎉 I'm happy to see that (apart from tests) it's mostly configuration and dependencies changes, with some minor tweaks in the code... however, at the same time it's a **massive** amount of work, which is truly tremendous 👏 🔥

loic425 commented on .github/workflows/application.yml:
```diff
@@ -24,22 +24,15 @@ jobs:
-                    -
-                        php: "8.1"
-                        symfony: "^5.4"
-                        api-platform: "~2.6.0"
```
why removing this api-platform checks?

Zales0123 replied:
When changing the ApiPlatform version there were problems with the validation of package versions (we would also need to change it inside the ApiBundle which seemed weird). I agree we should take care of this problem 👍 but maybe we could do it after the merge to not make iteration on this one even longer 😄

loic425 commented on config/packages/test/security.yaml:
```diff
@@ -1,7 +1,3 @@
-
-sylius_user:
-    encoder: sha512
+        Sylius\Component\User\Model\UserInterface: argon2i
```
this impacts performance a lot. why not using plainPassword for example.

Zales0123 replied:
Let's try to switch it in a separate PR, wdyt? It would be easier to track these performance changes (btw, it would be great if we finally have some performance tests in the pipeline 😄)

loic425 commented on src/Kernel.php:
```diff
@@ -61,13 +61,13 @@ protected function configureContainer(ContainerBuilder $container, LoaderInterfa
     }
 
-    protected function configureRoutes(RouteCollectionBuilder $routes): void
+    protected function configureRoutes(RoutingConfigurator $routes): void
```
we should remove these method

GSadee commented on .github/workflows/application.yml:
```diff
@@ -24,22 +24,15 @@ jobs:
-                        php: "8.1"
-                        symfony: "^5.4"
-                        api-platform: "~2.6.0"
+                symfony: ["^5.4", "^6.0"]
```
Shouldn't we test Symfony 6.0 and 6.1, or only the newest one?
```suggestion
                symfony: ["^5.4", "~6.0.0", "~6.1.0"]
```
Or maybe at least include one build for 6.0 🤔 
Of course here and in other workflows 😄

GSadee commented on psalm.xml:
```diff
@@ -22,12 +22,14 @@
             <directory name="src/Sylius/Bundle/ApiBundle/test" />
             <directory name="src/Sylius/Bundle/AdminBundle/spec" />
             <directory name="src/Sylius/Bundle/AdminBundle/Tests" />
+            <directory name="src/Sylius/Bundle/AdminBundle/test" />
```
We could probably add some more generic exclusion for these tests directories
```
<directory name="src/Sylius/Bundle/*/spec" />
<directory name="src/Sylius/Bundle/*/test" />
<directory name="src/Sylius/Bundle/*/Tests" />
```

loic425 commented on src/Sylius/Bundle/AddressingBundle/test/app/AppKernel.php:
```diff
@@ -32,17 +31,8 @@ public function registerBundles(): iterable
     }
 
-    public function registerContainerConfiguration(LoaderInterface $loader)
+    public function registerContainerConfiguration(LoaderInterface $loader): void
```
I think it can be removed too.

loic425 commented on src/Sylius/Bundle/AdminBundle/EventListener/ResourceDeleteSubscriber.php:
```diff
@@ -118,4 +122,13 @@ private function isAdminSection(array $syliusParameters): bool
         return array_key_exists('section', $syliusParameters) && 'admin' === $syliusParameters['section'];
     }
+
+    private function getFlashBag(): FlashBagInterface
```
Maybe it could be cool to add this on an Internal trait to simplify removing the bc-layer later.

loic425 commented on src/Sylius/Bundle/AdminBundle/Storage/FilterStorage.php:
```diff
@@ -13,26 +13,39 @@
+        return [] !== $this->getSession()->get('filters', []);
+    }
+
+    private function getSession(): SessionInterface
```
could be on an internal trait too.

loic425 commented on src/Sylius/Bundle/AdminBundle/test/src/Kernel.php:
```diff
@@ -0,0 +1,50 @@
+
+    private const CONFIG_EXTS = '.{php,xml,yaml,yml}';
+
+    public function getCacheDir(): string
```
can be removed

loic425 commented on src/Sylius/Bundle/AdminBundle/test/src/Kernel.php:
```diff
@@ -0,0 +1,50 @@
+        return $this->getProjectDir() . '/var/cache/' . $this->environment;
+    }
+
+    public function getLogDir(): string
```
this one too

loic425 commented on src/Sylius/Bundle/AdminBundle/test/src/Kernel.php:
```diff
@@ -0,0 +1,50 @@
+        return parent::getProjectDir() . '/test';
+    }
+
+    public function registerBundles(): iterable
```
can be removed too

GSadee commented on src/Sylius/Bundle/UserBundle/Provider/UserProviderInterface.php:
```diff
@@ -13,8 +13,10 @@
 
 interface UserProviderInterface extends BaseUserProviderInterface
 {
+    public function loadUserByUsername($username): UserInterface;
```
```suggestion
    public function loadUserByUsername(string $username): UserInterface;
```
👀

GSadee commented on src/Sylius/Bundle/UserBundle/Security/UserPasswordEncoder.php:
```diff
@@ -15,20 +15,39 @@
+            return;
+        }
+
+        throw new \InvalidArgumentException(
```
Shouldn't we trigger deprecation instead of throwing exception? 🤔

Zales0123 commented:
Thank you, Loïc, and everyone else involved in this tremendous effort 🚀 🔥 🏅
